hapi-fhir-jpaserver-mdm/src/main/java/ca/uhn/fhir/jpa/mdm/dao/MdmLinkFactory.java

```diff
@@ -37,7 +38,15 @@ public MdmLinkFactory(IMdmSettings theMdmSettings) {
 	 *
 	 * @return the new {@link MdmLink}
 	 */
-	public MdmLink newMdmLink() {
-		return new MdmLink(myMdmSettings.getRuleVersion());
+	public IMdmLink newMdmLink() {
+		// TODO check if mongo or jpa, return MDMLinkDocument or MDMLink accordingly
+		try {
+			IMdmLink mdmLink = myMdmSettings.getMdmLInkImpl().getDeclaredConstructor().newInstance();
```

Comment:
```suggestion
			IMdmLink mdmLink = myMdmLinkBuilder.newMdmLink();
```
This new bean will have a different implementation depending on the platform.

Reply:
So as per the conversation last time, the now calls another factory IMdmLinkImplFactory to build the MdmLink implementation. The reason I chose to do it this way is because MdmLink needs 2 attributes, the implementation and the mdm rules version. The problem with having two different MdmLinkFactory is mdm rules lives in the mdm module, which makes the wiring overly complicated, where as just having a factory calling another factory seems like the simpler solution